### DISCUSSION TITLE
HomeAssistante >Menu >Energy > Battery Storage > Sensors

### DISCUSSION THREAD

**Discussion by voodoopt** [mletenay/home-assistant-goodwe-inverter#67] [General]
Hi I have a problem integrating my Goodwe ES with Pylontech batteries in the Home assistant
I'm using the code that is given as an example

> sensor:
>   - platform: template
>     sensors:
>       # Template sensor for values of energy bought (active_power < 0)
>       energy_buy:
>         friendly_name: "Energy Buy"
>         unit_of_measurement: 'W'
>         value_template: >-
>           {% if states('sensor.goodwe_active_power')|float < 0 %}
>             {{ states('sensor.goodwe_active_power')|float * -1 }}
>           {% else %}
>             {{ 0 }}
>           {% endif %}
>       # Template sensor for values of energy sold (active_power > 0)
>       energy_sell:
>         friendly_name: "Energy Sell"
>         unit_of_measurement: 'W'
>         value_template: >-
>           {% if states('sensor.goodwe_active_power')|float > 0 %}
>             {{ states('sensor.goodwe_active_power')|float }}
>           {% else %}
>             {{ 0 }}
>           {% endif %}

But for the Battery

> sensor:
>   - platform: template
>     # Template sensor for values of Batery charge (active_power < 0)
>      battery_charge:
>       friendly_name: "Battery charge"
>       unit_of_measurement: 'W'
>       value_template: >-
>         {% if states('sensor.goodwe_pbattery')|float < 0 %}
>           {{ states('sensor.goodwe_pbattery')|float * -1 }}
>         {% else %}
>           {{ 0 }}
>         {% endif %}
>     # Template sensor for values of Batery discharge (active_power > 0)
>     battery_discharge:
>       friendly_name: "Battery discharge"
>       unit_of_measurement: 'W'
>       value_template: >-
>         {% if states('sensor.goodwe_pbattery')|float > 0 %}
>           {{ states('sensor.goodwe_pbattery')|float }}
>         {% else %}
>           {{ 0 }}
>         {% endif %}

But I'm in trouble. The home assistant doesn't show the sensors I create

**Comment by Darkezze** [comment]
Did you get any luck fixing it? I am having the same problem here but with another kind of battery, I suppose the problem is not the battery but the information provided from the Inversor, so, same ES series, same problem with showing the sensor in the energy dashboard.

**Comment by Xfire207** [comment]
same Problem here, goodwe GW3648-EM.  have idea ?
